app/src/main/java/org/kiwix/kiwixmobile/nav/destination/library/LocalLibraryFragment.kt

```diff
@@ -139,6 +148,62 @@ class LocalLibraryFragment : BaseFragment() {
     go_to_downloads_button_no_files.setOnClickListener {
       offerAction(FileSelectActions.UserClickedDownloadBooksButton)
     }
+
+    select_file.setOnClickListener {
+      showFileChooser()
+    }
+  }
+
+  private fun showFileChooser() {
+    val intent = Intent().apply {
+      action = Intent.ACTION_GET_CONTENT
+      type = "*/*"
+      addCategory(Intent.CATEGORY_OPENABLE)
+    }
+    try {
+      startActivityForResult(
+        Intent.createChooser(intent, "Select a zim file"),
+        FILE_SELECT_CODE
+      )
+    } catch (ex: ActivityNotFoundException) {
+      activity.toast(resources.getString(R.string.no_app_found_to_open), Toast.LENGTH_SHORT)
+    }
+  }
+
+  override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+    when (requestCode) {
+      FILE_SELECT_CODE -> {
+        data?.data?.let(::isValidFile)
+      }
+      else -> super.onActivityResult(requestCode, resultCode, data)
+    }
+  }
+
+  private fun isValidFile(uri: Uri) {
+    val filePath = FileUtils.getLocalFilePathByUri(
+      requireActivity().applicationContext, uri
+    )
+    if (filePath == null || !File(filePath).exists()) {
+      activity.toast(R.string.error_file_not_found)
+      return
+    }
+    val file = File(filePath)
+    if (!FileUtils.isValidZimFile(file.path)) {
+      activity.toast(R.string.error_file_invalid)
+      return
+    }
+    navigateToReaderFragment(file)
```

Comment:
Here we are validating the file and navigating to the reader fragment. `isValidFile` name doesn't satisfy the function also it's bad for testing. We should split two functionality into two functions and merge them into one.

Reply:
done